docs/source/output.md

```diff
@@ -0,0 +1,86 @@
+# Console Output
+
+By default, `cappa.parse` or `cappa.invoke` will internally construct an
+[Output](cappa.Output) object which controls the format of all console
+output, including how [Exit](cappa.Exit) and errors are formatted.
+
+Alternatively [cappa.parse](cappa.parse) and [cappa.invoke](cappa.invoke) both accept
+`Output` instances as an optional input, allowing you to customize the instance or
+otherwise make use of it before/after the CLI parse time.
+
+Any `Output` output will automatically make use of [rich](rich), and thus has access to
+its native styling syntax, and the ability to output native rich constructs.
+
+## Producing Output
+An [invoke](cappa.invoke) handler which depends on an argument annotated as `Output`
+will be automatically provided with the `Output` instance.
+
+```python
+import cappa
+from dataclasses import dataclass
+
+@cappa.command()
+def cli(output: Output):
+    output("Hi!") # e.g. output.output("Hi!")
+
+
+cappa.invoke(cli)
+
+# or
+
+@cappa.command(invoke=cli)
+@dataclass
+class CLI:
+    ...
+
+cappa.invoke(cli)
+```
+
+```{note}
+If you're familiar with `click`, `click.echo` is roughly equivalent to `Output.output`.
+```
+
+You can alternatively call `Output.error` to make use of error-specific default formatting/coloring
+
+## Exiting and Exit Codes
+
+Cappa provides an [Exit](cappa.Exit) class to allow someone to gracefully exit
+the program, without emitting a traceback.
+
+```python
+raise cappa.Exit(message="Oh no!", code=1)
+# or
+raise cappa.Exit("Oh no!", code=1)
+# or
+raise cappa.Exit(code=1)
+# or
+raise cappa.Exit("Graceful exit")  # i.e. status code 0!
+```
+
+If [Exit](cappa.Exit) is encountered anywhere within cappa's running context (e.g. an invoke handler,
+a parser, an action, etc), it will automatically `Output` the message using the method appropriate
+to the exit code (`code=0` -> `Output.output` and `code>0` -> `Output.error`), before gracefully exiting.
+
+## Customizing default formatting
+[cappa.parse](cappa.parse) and [cappa.invoke](cappa.invoke) both accept `Output`
+instances as an optional input, allowing you to customize default formatting.
+
+- output_format (default `{message}`): Exit code == 0
+- error_format (default `{short_help}\n\n[red]Error[/red]: {message}`): Exit
+  code != 0
+
+If, for example, you did not want to include `short_help` by default, and wanted
+"Error" to be orange, you could do as follows:
+
+```python
+output = cappa.Output(error_format="[orange]Error[/orange]: {message}")
+cappa.parse(Command, output=output)
+```
```

Comment:
I wanted to quickly test a different error formatting expanding on the example provided, but the blue error message still appears in red, I'm not sure what I'm doing wrong:

<img width="754" height="1072" alt="20250717_0706_754x1072_1752728813" src="https://github.com/user-attachments/assets/475ac279-21ff-4f50-bfd2-0ed4d65cbc2f" />


```
from dataclasses import dataclass

from cappa import Output, command, invoke


def cli(output: Output):
    output.output("Welcome to Cappa!")
    output.error("Don't panic it's the best python CLI library!")
    custom_output = Output(error_format="[blue]Error[/blue]: {message}")
    custom_output.error("This is a custom output format.")


@command(invoke=cli)
@dataclass
class CLI:
    ...

invoke(CLI)
```